hooks/stages/entrypoint_test.go

```diff
@@ -0,0 +1,156 @@
+package stages
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"net/http"
+	"testing"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/prebid/prebid-server/hooks/invocation"
+)
+
+func TestExecuteEntrypointHook_CannotModifyPayloadInsideHookCallMethod(t *testing.T) {
+	entrypointHooks = []EntrypointHook{mockNopEntrypointHook{}, mockNopEntrypointHook{}}
+
+	ctx := context.TODO()
+	body := []byte(`{"name": "John", "last_name": "Doe"}`)
+	reader := bytes.NewReader(body)
+	req, err := http.NewRequest(http.MethodPost, "https://prebid.com/openrtb2/auction", reader)
+	if err != nil {
+		t.Fatalf("Unexpected error creating http request: %s", err)
+	}
+
+	newBody, err := ExecuteEntrypointHook(ctx, invocation.Context{Endpoint: "auction", Timeout: time.Second}, req, body)
+	if err != nil {
+		t.Fatalf("Unexpected error executing entrypoint hook: %s", err)
+	}
+
+	if bytes.Compare(body, newBody) != 0 {
+		t.Errorf("Body changed inside hook.Call method, expected: %s, got: %s", string(body), string(newBody))
+	}
+}
+
+type mockNopEntrypointHook struct{}
+
+func (e mockNopEntrypointHook) Code() string {
+	return "MockedNopEntrypointHook"
+}
+
+func (e mockNopEntrypointHook) Call(ctx context.Context, invocationContext invocation.Context, payload ImmutableEntrypointPayload) (invocation.HookResult[MutableEntrypointPayload], error) {
+	_ = jsonparser.Delete(payload.Body, "name")
+	_, _ = jsonparser.Set(payload.Body, []byte(`"bar"`), "foo")
```

Comment:
Please avoid using jsonparser Delete and Set methods. They are indicated as not production ready by the library author. We've tried to use them in production before and encountered situations where they panic.

They should be ok for tests, but might give the wrong impression that we as the PBS project support the use. We strictly use jsonparser for read only operations.